src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/NonAssociationsService.kt

```diff
@@ -36,4 +42,18 @@ class NonAssociationsService(
   private fun persistNonAssociation(nonAssociation: NonAssociationJPA): NonAssociationJPA {
     return nonAssociationsRepository.save(nonAssociation)
   }
+
+  fun mergePrisonerNumbers(oldPrisonerNumber: String, newPrisonerNumber: String): List<NonAssociationJPA> {
```

Comment:
this doesn't persist the changes does it?

Reply:
actually, even when it does, there could be a separate problem where the newly created copy matches a non-association pair in the DB. imagine:
• A has a non-association to B and another to C
• B is merged into C
• now we have two non-associations between A and C

Reply:
Yes I did think of this, its needs more work to handle that!

Reply:
Also the persistant bit was missing which I added this morning- sorry remember after the commit yesterday!